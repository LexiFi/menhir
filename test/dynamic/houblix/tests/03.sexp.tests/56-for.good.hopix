let x : mut<int> = ref (4)

let main : int =
    for i in (0 to 30) {
      x := !x + 1
    };
    !x