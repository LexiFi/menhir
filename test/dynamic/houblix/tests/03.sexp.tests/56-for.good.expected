(((value
    (DefineValue
      (SimpleValue
        (((value (Id x))
           (position ((start_p ("" 1 0 4)) (end_p ("" 1 0 5)))))
          (((value
              (ForallTy ()
                ((value
                   (TyCon (TCon mut)
                     (((value (TyCon (TCon int) ()))
                        (position
                          ((start_p ("" 1 0 12)) (end_p ("" 1 0 15))))))))
                  (position ((start_p ("" 1 0 8)) (end_p ("" 1 0 16)))))))
             (position ((start_p ("" 1 0 7)) (end_p ("" 1 0 16))))))
          ((value
             (Ref
               ((value
                  (Literal
                    ((value (LInt 4))
                      (position ((start_p ("" 1 0 24)) (end_p ("" 1 0 25)))))))
                 (position ((start_p ("" 1 0 23)) (end_p ("" 1 0 26)))))))
            (position ((start_p ("" 1 0 19)) (end_p ("" 1 0 26)))))))))
   (position ((start_p ("" 1 0 0)) (end_p ("" 1 0 26)))))
  ((value
     (DefineValue
       (SimpleValue
         (((value (Id main))
            (position ((start_p ("" 3 28 32)) (end_p ("" 3 28 36)))))
           (((value
               (ForallTy ()
                 ((value (TyCon (TCon int) ()))
                   (position ((start_p ("" 3 28 39)) (end_p ("" 3 28 42)))))))
              (position ((start_p ("" 3 28 38)) (end_p ("" 3 28 42))))))
           ((value
              (Sequence
                (((value
                    (For
                      ((value (Id i))
                        (position
                          ((start_p ("" 4 45 53)) (end_p ("" 4 45 54)))))
                      ((value
                         (Literal
                           ((value (LInt 0))
                             (position
                               ((start_p ("" 4 45 59)) (end_p ("" 4 45 60)))))))
                        (position
                          ((start_p ("" 4 45 59)) (end_p ("" 4 45 60)))))
                      ((value
                         (Literal
                           ((value (LInt 30))
                             (position
                               ((start_p ("" 4 45 64)) (end_p ("" 4 45 66)))))))
                        (position
                          ((start_p ("" 4 45 64)) (end_p ("" 4 45 66)))))
                      ((value
                         (Assign
                           ((value
                              (Variable
                                ((value (Id x))
                                  (position
                                    ((start_p ("" 5 70 76))
                                      (end_p ("" 5 70 77)))))
                                ()))
                             (position
                               ((start_p ("" 5 70 76)) (end_p ("" 5 70 77)))))
                           ((value
                              (Apply
                                ((value
                                   (Apply
                                     ((value
                                        (Variable
                                          ((value (Id `+`))
                                            (position
                                              ((start_p ("" 5 70 84))
                                                (end_p ("" 5 70 85)))))
                                          ()))
                                       (position
                                         ((start_p ("" 5 70 84))
                                           (end_p ("" 5 70 85)))))
                                     ((value
                                        (Read
                                          ((value
                                             (Variable
                                               ((value (Id x))
                                                 (position
                                                   ((start_p ("" 5 70 82))
                                                     (end_p ("" 5 70 83)))))
                                               ()))
                                            (position
                                              ((start_p ("" 5 70 82))
                                                (end_p ("" 5 70 83)))))))
                                       (position
                                         ((start_p ("" 5 70 81))
                                           (end_p ("" 5 70 83)))))))
                                  (position
                                    ((start_p ("" 5 70 81))
                                      (end_p ("" 5 70 85)))))
                                ((value
                                   (Literal
                                     ((value (LInt 1))
                                       (position
                                         ((start_p ("" 5 70 86))
                                           (end_p ("" 5 70 87)))))))
                                  (position
                                    ((start_p ("" 5 70 86))
                                      (end_p ("" 5 70 87)))))))
                             (position
                               ((start_p ("" 5 70 81)) (end_p ("" 5 70 87)))))))
                        (position
                          ((start_p ("" 5 70 76)) (end_p ("" 5 70 87)))))))
                   (position ((start_p ("" 4 45 49)) (end_p ("" 6 88 93)))))
                  ((value
                     (Read
                       ((value
                          (Variable
                            ((value (Id x))
                              (position
                                ((start_p ("" 7 95 100))
                                  (end_p ("" 7 95 101)))))
                            ()))
                         (position
                           ((start_p ("" 7 95 100)) (end_p ("" 7 95 101)))))))
                    (position ((start_p ("" 7 95 99)) (end_p ("" 7 95 101))))))))
             (position ((start_p ("" 4 45 49)) (end_p ("" 7 95 101)))))))))
    (position ((start_p ("" 3 28 28)) (end_p ("" 7 95 101))))))
